flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingMetricEvaluator.java

```diff
@@ -341,6 +375,37 @@ public static double getAverage(
         return getAverage(metric, jobVertexId, metricsHistory, 1);
     }
 
+    public static double getRate(
+            ScalingMetric metric,
+            @Nullable JobVertexID jobVertexId,
+            SortedMap<Instant, CollectedMetrics> metricsHistory) {
+
+        Instant firstTs = null;
+        double first = Double.NaN;
+
+        Instant lastTs = null;
+        double last = Double.NaN;
+
+        for (var entry : metricsHistory.entrySet()) {
+            double value = entry.getValue().getVertexMetrics().get(jobVertexId).get(metric);
+            if (!Double.isNaN(value)) {
+                if (Double.isNaN(first)) {
+                    first = value;
+                    firstTs = entry.getKey();
+                } else {
+                    last = value;
+                    lastTs = entry.getKey();
+                }
+            }
+        }
+        if (Double.isNaN(last)) {
+            return Double.NaN;
+        }
+
+        double diff = last - first;
+        return diff == 0 ? 0 : diff / Duration.between(firstTs, lastTs).toSeconds();
```

Comment:
I will this, I still had to change the logic so that we compute the seconds rate using the millisecond diff otherwise we can accidentally divide by zero